elasticsearch_dsl/_async/document.py

```diff
@@ -69,12 +80,23 @@ class AsyncDocument(DocumentBase, metaclass=AsyncIndexMeta):
     Model-like class for persisting documents in elasticsearch.
     """
 
+    if TYPE_CHECKING:
+        _index: AsyncIndex
+
     @classmethod
-    def _get_connection(cls, using=None):
+    def _get_using(cls, using: Optional[AsyncUsingType] = None) -> AsyncUsingType:
```

Comment:
_get_using() was in the `DocumentBase` class, but now that it needs to be typed it needs to come down to the async/sync specialized classes.